dynamic_programming/range_maximum_query.py

```diff
@@ -0,0 +1,134 @@
+"""
+Range Maximum Query Problem:
+
+    Given an array of integers and q queries,
+    For each query find the maximum value in the range [l, r).
+
+    The array is 0-indexed and the queries are 0-indexed.
+
+    More details:
+        https://www.geeksforgeeks.org/range-maximum-query-using-sparse-table/
+"""
+from __future__ import annotations
+
+
+class RangeMaximumQuery:
+    def __init__(self, array: list | tuple) -> None:
+        """
+        Initialize RangeMaximumQuery with given array.
+
+        Parameters:
+            array: list[int]
+
```

Comment:
We can remove these lines especially given that they disagree with the type hints a few lines above.